### ISSUE TITLE
Unable to send tokens which are available only on L1 network

### ISSUE THREAD

**Issue by pavloburykh** [status-im/status-react#21292]
Fro reproduction you can use SNT

**Steps:** 

1. Restore the account which has SNT 
2. Proceed with sending flow
3. See if the routes are built after entering valid amount

**Actual result:** 

[Status-debug-logs - 2024-09-19T171851.912.zip](https://github.com/user-attachments/files/17060939/Status-debug-logs.-.2024-09-19T171851.912.zip)

On mobile I receive different errors, like ```transfer amount exceeds balance``` or ```not available for chain ID```

![photo_2024-09-19 17 04 50](https://github.com/user-attachments/assets/d15db506-2095-4f65-b23e-e6abddf20017)
![photo_2024-09-19 17 04 55](https://github.com/user-attachments/assets/7bfc8cc2-a44d-488f-940a-6d497df5ccfc)

Same on Desktop: 

<img width="542" alt="Status Desktop 2024-09-19 17-09-01" src="https://github.com/user-attachments/assets/5b2b3eec-36bd-4189-9cf4-fdf676892d2e">

cc @saledjenic @anastasiyaig 
### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

**Comment by anastasiyaig** [comment]
i was able to send snt just 2 days ago. i dont have any context besides that. maybe something got broken recently 

@virginiabalducci can you pls check?

**Comment by balducciv** [comment]
I was able to do a send tx in testnet but I'm reproducing the same issue in mainnet
![Screenshot 2024-09-19 at 11 28 57 AM](https://github.com/user-attachments/assets/2f0e6cda-2623-4521-abb7-5e843d36a937)

https://github.com/user-attachments/assets/0112d279-f6ec-45c9-9eab-958dbf00f5c4
[app_20240917_124336.log](https://github.com/user-attachments/files/17061105/app_20240917_124336.log)

If needed, adding addresses I tested with:
From: 0x32e12fd7c2c7984dd6de153691d901445fc8aa07
To: 0x69af29dedbe58f4001bcf14803963cb7894112b3

**Comment by anastasiyaig** [comment]
thanks @virginiabalducci 
@saledjenic @dlipicar do you know the root cause?

**Comment by saledjenic** [comment]
@virginiabalducci I was able to reproduce your case with `0x32e12fd7c2c7984dd6de153691d901445fc8aa07` address.

So let's analyze it:
- you want to send 1 SNT from the `0x32e12fd7c2c7984dd6de153691d901445fc8aa07` from any of the supported chains (mainnet, opt, arb) to the recipient address to any chain (mainnet, opt, arb) 
- what are possible options (since you have a positive balance for SNT only on mainnet):
  1. mainnet -> mainnet (transfer)
  2. mainnet -> optimism (bridge)
  3. mainnet -> arbitrum (bridge)
- let's look now at how the router sees that request since all chains are available on both from and to sides:
  1. check transfer for mainnet -> mainnet -> this returns `insufficient funds for gas * price + value: address 0x32e12fD7C2C7984DD6de153691D901445fC8AA07 have 16773029697747188 want 235096461785385998`
  2. check transfer optimism -> optimism (transfer) -> `execution reverted: ERC20: transfer amount exceeds balance` cause you have 0 SNT on optimism
  3. check transfer arbitrum -> arbitrum (transfer) -> `execution reverted: ERC20: transfer amount exceeds balance` cause you have 0 SNT on arbitrum
  4. check bridge `mainnet` -> `optimism` -> `not available for chainID` means there is no contract for bridging SNT
  5. check bridge `mainnet` -> `arbitrum` -> `not available for chainID` means there is no contract for bridging SNT
  6. check bridge `optimism` -> `mainnet` -> `not available for chainID` means there is no contract for bridging SNT
  7. check bridge `optimism` -> `arbitrum` -> `not available for chainID` means there is no contract for bridging SNT
  8. check bridge `arbitrum` -> `mainnet` -> `not available for chainID` means there is no contract for bridging SNT
  9. check bridge `arbitrum` -> `optimism` -> `not available for chainID` means there is no contract for bridging SNT
- now looking at the previous point we confirmed that there is no route that can be made, which is correct, that's what the router says, but a question is about the error, which one of those is the most meaningful for you? You cannot choose one, but the right way would be to return all of them, right? Displaying all of them was rejected as an idea in one of my PRs and then we had to choose one to display. There is some logic for selecting the most meaningful error which is working quite well, but in case like this one where all chains are in play on the "from" and "to" side is hard to select one (especially when we have different processor orders among router runs) cause all are valid errors.

**To summarize and conclude, there is no error in the router logic.**
@virginiabalducci @anastasiyaig @pavloburykh @alwx

**Comment by pavloburykh** [comment]
> **To summarize and conclude, there is no error in the router logic.**
> @virginiabalducci @anastasiyaig @pavloburykh @alwx

Thank you for the comment @saledjenic 

I have couple of questions here:

1. Regarding insufficient funds error:  we have an open issue https://github.com/status-im/status-mobile/issues/20970. Could you please leave a comment in that issue summarising your findings regrading this problem. I remember that recently there have been much discussions in different Discord channels, but unfortunately there is no summary in the ticket. Currently we are not able to perform L1-L1 transactions in Status app, at the same time I am able to perform same transaction with the same account using Metamask. 


2. At which point do we understand that token is available only on L1 and thus there is no sense to try building any routes except L1-L1? Can we prevent from trying to perform ```mainnet -> optimism (bridge)```, ```mainnet -> arbitrum (bridge)``` etc steps if we know that only mainnet-mainnet is available? To be honest, this is very confusing that we are showing user  the irrelevant errors like ```transfer amount exceeds balance``` or ```not available for chain ID``` while app UI shows that we are trying to build L1-L1 route and both errors have nothing to do with L1-L1 in this case. Why we are not showing ```insufficient funds``` here which is related to L1-L1 route building?

**Comment by churik** [comment]
I believe we can close https://github.com/status-im/status-mobile/issues/21292 in favour of https://github.com/status-im/status-mobile/issues/20970 and wait for new RPC provider integration 🤞

**Comment by anastasiyaig** [comment]
i would suggest to check proxy logic  since the issue was not on the router side, but on proxy. So querying Infura directly was just fine. Replacing provider wont help in this specific case. There is an explanation in https://github.com/status-im/status-mobile/issues/20970 @churik

**Comment by churik** [comment]
There is no actually one conclusion (as it was checked with disabled both grove and proxy), and the discussion moved there, but yeah, seems that the issue might be on both sides - grove and proxy. 
Thanks for the comment. I'll bring this issue today on the wallet sync
